src/ESPiLight.cpp

```diff
@@ -443,3 +448,76 @@ int ESPiLight::stringToPulseTrain(const String &data, uint16_t *codes,
   }
   return -1;
 }
+
+static protocols_t *find_proto(const char *name) {
+  struct protocols_t *pnode = protocols;
+  while (pnode != nullptr) {
+    if (strcmp(name, pnode->listener->id) == 0) {
+      return pnode;
+    }
+    pnode = pnode->next;
+  }
+  return nullptr;
+}
+
+void ESPiLight::limitProtocols(const String &protos) {
+  if (!json_validate(protos.c_str())) {
+    DebugLn("Protocol limit argument is not a valid json message!");
+    return;
+  }
+  JsonNode *message = json_decode(protos.c_str());
+
+  if (message->tag != JSON_ARRAY) {
+    DebugLn("Protocol limit argument is not a json array!");
+    json_delete(message);
+    return;
+  }
+
+  if (used_protocols != protocols) {
+    struct protocols_t *pnode = used_protocols;
+    while (pnode != nullptr) {
+      struct protocols_t *tmp = pnode;
+      pnode = pnode->next;
+      delete tmp;
+    }
+  }
+
+  used_protocols = nullptr;
+  JsonNode *curr = message->children.head;
+  unsigned int proto_count = 0;
+
+  while (curr != nullptr) {
+    if (!curr->tag == JSON_STRING) {
+      DebugLn("Element is not a String");
+      continue;
+    }
+
+    protocols_t *templ = find_proto(curr->string_);
```

Comment:
Somewhere we use `struct` , somewhere not. It is not required for C++, so we can completely rid of it?